Added HW 10 (Serialization)

IhorStetskyi opened the pull request.

IhorStetskyi committed:
Added HW 10 (Serialization)

Hekitkoo commented on Stetskyi_Homework_10/Homework10/Sources/CustomSerializer.cs:
```diff
@@ -0,0 +1,83 @@
+        {
+            _type = type;
+        }
+        public SerializationBinder Binder { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
```
@IhorStetskyi, I believe that we should avoid using NotImplementedException.

IhorStetskyi committed:
- System.Text.Json.JsonSerializer part fixed
- changed local variables aka pascalCase
- path strings changed to const and moved up to the Class entity
- fields changed to properties
- "not implemented" removed from CustomSerializer
- IXmlSerializable implemented

IhorStetskyi committed:
- additional fixes regarding PascalCase for constants & camelCase for local variables
- [JsonConstructor] used for System.Text.Json.JsonSerializer
